Merge command failed with:

```
2025-11-11 11:22:07,202 271 INFO odoo odoo.addons.pos_user_restriction.tests.test_hacks: Starting TestHacks.test_validate_session ... 
2025-11-11 11:22:07,304 271 INFO odoo odoo.models.unlink: User #18 deleted pos.payment.method records with IDs: [27] 
2025-11-11 11:22:07,346 271 ERROR odoo odoo.sql_db: bad query: INSERT INTO "account_journal" ("active", "bank_statements_source", "check_manual_sequencing", "code", "color", "company_id", "create_date", "create_uid", "default_account_id", "invoice_reference_model", "invoice_reference_type", "loss_account_id", "name", "payment_sequence", "profit_account_id", "refund_sequence", "sale_activity_type_id", "sequence", "show_on_dashboard", "type", "write_date", "write_uid") VALUES (true, 'undefined', false, 'CSH 1', 0, 8, '2025-11-11 11:22:01.244164', 18, 375, 'odoo', 'invoice', 344, '{"en_US": "Cash"}', true, 334, false, NULL, 10, true, 'cash', '2025-11-11 11:22:01.244164', 18) RETURNING "id"
ERROR: duplicate key value violates unique constraint "account_journal_code_company_uniq"
DETAIL:  Key (company_id, code)=(8, CSH 1) already exists.
 
2025-11-11 11:22:07,349 271 INFO odoo odoo.addons.pos_user_restriction.tests.test_hacks: ====================================================================== 
2025-11-11 11:22:07,349 271 ERROR odoo odoo.addons.pos_user_restriction.tests.test_hacks: ERROR: TestHacks.test_validate_session
Traceback (most recent call last):
  File "/__w/pos/pos/pos_user_restriction/tests/test_hacks.py", line 49, in test_validate_session
    self.config = self._create_basic_config()
  File "/opt/odoo/addons/point_of_sale/tests/common.py", line 269, in _create_basic_config
    'journal_id': cls.env['account.journal'].create({
  File "<decorator-gen-224>", line 2, in create
  File "/opt/odoo/odoo/api.py", line 430, in _model_create_multi
    return create(self, [arg])
  File "/opt/odoo/addons/account_check_printing/models/account_journal.py", line 62, in create
    journals = super().create(vals_list)
  File "<decorator-gen-209>", line 2, in create
  File "/opt/odoo/odoo/api.py", line 431, in _model_create_multi
    return create(self, arg)
  File "/opt/odoo/addons/account/models/account_journal.py", line 828, in create
    journals = super(AccountJournal, self.with_context(mail_create_nolog=True)).create(vals_list)
  File "<decorator-gen-140>", line 2, in create
  File "/opt/odoo/odoo/api.py", line 431, in _model_create_multi
    return create(self, arg)
  File "/opt/odoo/addons/mail/models/mail_alias_mixin_optional.py", line 92, in create
    records = super().create(valid_vals_list)
  File "<decorator-gen-142>", line 2, in create
  File "/opt/odoo/odoo/api.py", line 431, in _model_create_multi
    return create(self, arg)
  File "/opt/odoo/addons/mail/models/mail_thread.py", line 255, in create
    threads = super(MailThread, self).create(vals_list)
  File "<decorator-gen-12>", line 2, in create
  File "/opt/odoo/odoo/api.py", line 431, in _model_create_multi
    return create(self, arg)
  File "/opt/odoo/odoo/models.py", line 4656, in create
    records = self._create(data_list)
  File "/opt/odoo/odoo/models.py", line 4844, in _create
    cr.execute(SQL(
  File "/opt/odoo/odoo/sql_db.py", line 335, in execute
    res = self._obj.execute(query, params)
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "account_journal_code_company_uniq"
DETAIL:  Key (company_id, code)=(8, CSH 1) already exists.
```


Can you rebase and see if the issue is reproduced here?

--- a/pos_user_restriction/models/pos_session.py
+++ b/pos_user_restriction/models/pos_session.py
@@ -10,3 +10,17 @@ class PosSession(models.Model):
         ):
             self = self.with_context(bypass_pos_user=True)
         return super().get_closing_control_data()
+
+    def _validate_session(
+        self,
+        balancing_account=False,
+        amount_to_balance=0,
+        bank_payment_method_diffs=None,
+    ):
+        if self.env.user.has_group(
+            "pos_user_restriction.group_assigned_points_of_sale_user"
+        ):
+            self = self.with_context(bypass_pos_user=True)
+        return super()._validate_session(
+            balancing_account, amount_to_balance, bank_payment_method_diffs
+        )

--- a/pos_user_restriction/tests/test_hacks.py
+++ b/pos_user_restriction/tests/test_hacks.py
@@ -32,13 +32,67 @@ class TestHacks(TestPoSCommon):
                 ],
             }
         )
+        cls.config = cls.basic_config
 
     def test_get_closing_control_data(self):
         restricted_user = self.pos_user_assigned_pos
-        self.config = self._create_basic_config()
         self.config.assigned_user_ids = [(6, 0, [restricted_user.id])]
 
         session = self.open_new_session()
 
         # make sure it does raise AccessError
         session.with_user(restricted_user).get_closing_control_data()
+
+    def test_validate_session(self):
+        restricted_user = self.pos_user_assigned_pos
+
+        self.config.assigned_user_ids = [(6, 0, [restricted_user.id])]
+        self.product_id = self.env["product.product"].create(
+            {"name": "Test POS", "available_in_pos": True, "list_price": 200}
+        )
+
+        session = self.open_new_session()
+        self.pos_order_pos1 = self.env["pos.order"].create(
+            {
+                "company_id": self.env.company.id,
+                "session_id": session.id,
+                "partner_id": self.env.user.id,
+                "lines": [
+                    (
+                        0,
+                        0,
+                        {
+                            "name": "Test/0001",
+                            "product_id": self.product_id.id,
+                            "price_unit": 200,
+                            "qty": 1.0,
+                            "price_subtotal": 200,
+                            "price_subtotal_incl": 200,
+                        },
+                    )
+                ],
+                "amount_tax": 0.0,
+                "amount_total": 200,
+                "amount_paid": 0.0,
+                "amount_return": 0.0,
+                "last_order_preparation_change": "{}",
+            }
+        )
+        context_make_payment = {
+            "active_ids": [self.pos_order_pos1.id],
+            "active_id": self.pos_order_pos1.id,
+        }
+        self.pos_make_payment = (
+            self.env["pos.make.payment"]
+            .with_context(**context_make_payment)
+            .create(
+                {
+                    "amount": 200,
+                }
+            )
+        )
+        context_payment = {"active_id": self.pos_order_pos1.id}
+        self.pos_make_payment.with_context(**context_payment).check()
+
+        # make sure it doesn't raise AccessError
+        session.with_user(restricted_user).action_pos_session_closing_control()

